src/graph/DependencyGraph.cpp

```diff
@@ -82,6 +82,56 @@ size_t DependencyGraph::addNode()
   return roots_.size() - 1;
 }
 
+void DependencyGraph::removeNode(size_t id)
+{
+  using Edge = std::pair<size_t, size_t>;
+  assert(id < roots_.size());
+  // Remove all edges from/to {id}
+  for(auto it = edges_.begin(); it != edges_.end();)
+  {
+    const Edge & e = *it;
+    if(e.first == id || e.second == id)
+    {
+      it = edges_.erase(it);
+    }
+    else
+    {
+      ++it;
```

Comment:
This should also decrement the edge `from` (resp. `to`) if `from > id` (resp. `to`)

Reply:
Fixed in cebeda8379ebf7d9d8be1b6a8d00322f692bf440